app/src/main/java/com/yujeong/aostudy/broccoli/ui/base/BaseFragment.kt

```diff
@@ -0,0 +1,30 @@
+package com.yujeong.aostudy.broccoli.ui.base
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.fragment.app.Fragment
+
+abstract class BaseFragment<T : ViewDataBinding>(@LayoutRes private val layoutResId : Int) : Fragment() {
+    private var _binding : T? = null
+    val binding get() = _binding ?: error("View를 참조하기 위해 binding이 초기화되지 않았습니다.")
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = DataBindingUtil.inflate(inflater, layoutResId, container, false)
+        binding.lifecycleOwner = this
```

Comment:
this -> Fragment 자체의 생명주기를 참조하게 되는것! -> ViewPager ㅁ ㅁ ㅁ onDestroy까지는 가지않음 View자체는 화면에 보이지 않지만, Fragment가 계속 살아있어서, lifecycleOwner가 프래그먼트 그 자체라서, 프래그먼트의 화면은 꺼져서 다른 화면이 보여서 더이상 참조할 필요가 없는데도 계속 참조를 하게 되는 그런 문제가 발생을 해 -> LiveData -> observer를 달 때 저 라이프사이클 오너란 걸 쓰는데,

```suggestion
        binding.lifecycleOwner = viewLifeCycleOwner
```